src/screens/EditorScreen.tsx

```diff
@@ -280,6 +293,15 @@ export function EditorScreen() {
         }
     }, [currentContent, hasFileOpen, selection?.path]);
 
+    // Finds the correct editor/viewer component given the current file type
+    const FileEditor = useMemo(() => {
+        return (selection?.path && Object.entries(FILE_COMPONENTS).find(([pattern, Component]) => {
+            const regex = new RegExp(`^.*\\.(${pattern})$`, "i");
+            if (regex.test(selection.path ?? "")) {
+                return true;
+            }
+        })?.[1]) || TextEditor;
+    }, [selection?.path]);
```

Comment:
This might actually be a case where a switch on file extension with multiple cases for ImageViewer and TextEditor as default is more clear.